app/models/oauth_client.rb

```diff
@@ -0,0 +1,25 @@
+class OauthClient < ApplicationRecord
+  validates :client_id, presence: true, uniqueness: true
+  validates :client_name, presence: true
+  validates :redirect_uri, presence: true
+
+  def valid_redirect_uri?(uri)
```

Comment:
I'm not sure if this is superior to just doing the `==` comparison as part of the `validate_authorization_request` method. It implies that there is more to being a valid redirect URI than just matching the saved value when that's not currently the case; will more complex validations be introduced later?

Reply:
I see yeah that makes sense, and I don't have any more validations to add, at least not atm. Until i do, I agree that this would condense the code and keep it concise. I'll remove this from the model and make the comparison in that controller method you suggested above instead.